### PR TITLE
feat(cli): orchestration backend setting, mcp update, --name lookups (#913)

### PR DESCRIPTION
**Pull request by calumjs**
## Problem

The `yakshaver` CLI could read but not fully configure the desktop app. This PR adds a localhost-only, token-authenticated bridge surface so the CLI can set the orchestration backend, update/enable/remove MCP servers (including by `--name`), and read/write user settings — without ever handling or echoing secrets.

## What changed

- **Orchestration backend**: `config get|set orchestrator` reads/sets `LLMConfigV2.orchestrationBackend` (`openai` | `local-claude`). `config get` always surfaces a concrete value, defaulting to `openai` — including on a fresh install where no LLM config exists yet.
- **MCP management**: `mcp add|update|remove|enable`, with `--name <name>` selectors resolved against the server list. `update` is a merge-patch (only provided fields change). A transport switch (HTTP <-> stdio) normalizes to the target discriminated-union member and strips the old transport's fields so no stale `url`/`headers` or `command`/`args`/`env` survive — both at the bridge and in the manager's own persistence merge.
- **Built-in servers** (always-on, in-memory) cannot be modified or removed via the CLI; such requests are rejected with a clear error instead of silently no-op'ing.
- **stdio args**: a repeatable `--arg <value>` flag preserves each argument verbatim (so a value may contain spaces, e.g. `--arg "/My Documents/server.js"`). The legacy whitespace-split `--args "a b c"` is kept for back-compat; the two are mutually exclusive.
- **Settings**: `config get|set settings` for `--tool-approval-mode` and `--open-at-login`.

## Security model

- The bridge binds to `127.0.0.1` only (never `0.0.0.0`).
- A random 256-bit bearer token is generated at startup and written to `userData/yakshaver-tokens/cli-bridge.json` (owner-readable where the OS honours it); every request must present `Authorization: Bearer <token>`.
- All secrets (api keys, header/env values) are redacted in every response.
- All write payloads are validated with strict zod schemas (`LLMConfigV2Schema`, MCP input/patch schemas) before persistence.

## Test plan

- `npm run build` (tsc) — passes.
- `npx vitest run` over `src/cli`, `src/backend/services/cli-bridge`, `src/backend/services/mcp` — 120 tests pass, including:
  - repeatable `--arg` preserving spaces + rejecting `--arg`/`--args` mixing;
  - built-in server enable/update/remove rejection;
  - fresh-install (null config) `config get orchestrator` -> `openai`;
  - an integration test driving the transport switch through the REAL `MCPServerManager.updateServerAsync` (storage mocked) asserting no stale secret-bearing fields persist.
- `npx biome check` — clean on all touched files.

(DB and ffmpeg suites are skipped here due to native-binding/platform limitations in the CI worktree, unrelated to this change.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by calumjs — COMMENTED** [review]
## Review: #913 CLI orchestrator setting

Four findings below, grouped by severity. The two HIGH items are the priority: one is a correctness bug that degrades the #833 outcome judge, the other is a PR-stacking hazard that risks merging an unreviewed copy of #908.

### HIGH

**1. tool_result records the opaque `tool_use_id` as `toolName`, not the real tool name** — `src/backend/services/mcp/local-claude-orchestrator.ts:435`

In `handleStreamEvent`, the `user`/`tool_result` branch records `toolName: block.tool_use_id ?? "unknown"`. In Claude Code's `stream-json` (raw Anthropic content blocks), a tool_result's `tool_use_id` is the **opaque correlation id** of the originating tool_use block (e.g. `toolu_01ABC...`), not the human-readable name like `mcp__GitHub__create_issue`. The real name is on the assistant `tool_use` block (`block.name`), which is used only for the ephemeral `onStep` event and then discarded — there is no id->name map.

Consequence: at real runtime every `toolActivity[].toolName` becomes a `toolu_...` id. That array is JSON-stringified into the LLM judge prompt as the success/fail "ground truth" (`backlog-orchestrator.ts`) and printed in the `tools:` diagnostic line, so both the judge and the logs lose which tool actually ran — directly weakening the #833 outcome decision this PR is built around. This also diverges from the OpenAI twin, which correctly records `toolName: toolCall.toolName` (`mcp-orchestrator.ts`).

The unit test masks this: the fixture sets `tool_use_id` to a tool-name-shaped string (`mcp__yakshaver__GitHub__create_issue`) — a value a real CLI never emits — and only asserts the `onStep` toolName (the working `block.name` path), never the recorded `toolActivity[].toolName`.

Remedy: build a `tool_use_id -> name` map from the assistant `tool_use` blocks, then resolve `toolName` on the tool_result from that map (fall back to the id only if unmapped). Update the fixtures to use a realistic opaque `toolu_01...` id on both the tool_use `id` and the tool_result `tool_use_id`, and assert the recorded `toolName` is the human-readable name — so the correlation is actually exercised.

**2. PR bundles the entire #908 orchestration backend as net-new files** — base branch `feat/909-yakshaver-cli`

The diff adds the whole #908 backend (`local-claude-orchestrator.ts`, `backlog-orchestrator.ts`, the `mcp-orchestrator.ts` judge extraction, `OrchestratorBackendSetting.tsx`, `llm.ts` orchestrationBackend field — 8 files) as brand-new files, even though that code is the separate open PR #908 (base `main`, issue #907). #908's head is an ancestor of #914's head, but the PR base `feat/909-yakshaver-cli` does not contain those files, so they show as ADDED and the actual #913 increment is co-mingled with all of #908. Issue #913 itself says "Stacked on #908 + #910 — merge #908 and #910 first, then this."

If #909 merges to main before #908, or #908 is reworked/rejected during its own review, merging #914 silently pulls in a stale/unreviewed copy of #908's backend, and the two copies can drift.

Remedy: re-target #914's base to #908's branch (`feat/907-claude-code-orchestration`) so the diff shows only the #913 increment (bridge route + CLI commands + name resolver), or hold #914 until #908/#910 land on main and rebase onto main.

### MEDIUM

**3. MCP config containing live OAuth bearer tokens is written to the shared temp dir in plaintext** — `src/backend/services/mcp/local-claude-orchestrator.ts:142` (token injected at `:268`)

`toClaudeServerEntry()` injects `Authorization: Bearer ${tokens.access_token}` for each non-builtin streamableHttp server, then `manualLoopAsync()` writes the serialized config to `join(tmpdir(), ...)` via `fs.writeFile(..., JSON.stringify(mcpConfig, null, 2), "utf8")` with no `mode` — default permissions in the OS-shared temp dir. Cleanup is `fs.unlink` in a `finally` only, so a SIGKILL leaves the plaintext token on disk.

This bypasses the app's own at-rest convention: every other secret (OAuth tokens, GitHub/identity tokens, LLM config) is encrypted via Electron `safeStorage` into `userData` (`base-secure-storage.ts`), with code that throws if OS encryption is unavailable. The local-orchestrator path writes the same tokens in cleartext to a predictable, world-readable location.

Remedy: pass the config inline — `--mcp-config` accepts a JSON string, so the token never touches disk — or, if a file is required, write with `{ mode: 0o600 }` and ensure cleanup on process exit/signal as well as in `finally`.

File: src/backend/services/mcp/local-claude-orchestrator.ts [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    }
+
+    // User turn carries tool_result blocks (Claude reports tool outputs as a user message).
+    if (event.type === "user" && event.message?.content) {
+      for (const block of event.message.content) {
+        if (block.type === "tool_result") {
+          const resultText = extractToolResultText(block.content);
+          const ok = block.is_error !== true;
+          toolActivity.push({
+            toolName: block.tool_use_id ?? "unknown",
```
Comment (calumjs) [inline src/backend/services/mcp/local-claude-orchestrator.ts]: HIGH: `block.tool_use_id` is the opaque correlation id (`toolu_01...`), not the tool name. At real `stream-json` runtime this stores ids like `toolu_...` into `toolActivity[].toolName`, which is then fed to the #833 outcome judge as ground truth and printed in the `tools:` log line — losing which tool ran. The real name is on the assistant `tool_use` block (`block.name`), discarded here. Build a `tool_use_id -> name` map from the assistant tool_use blocks and resolve `toolName` from it (fall back to the id only if unmapped). The OpenAI twin already does this via `toolCall.toolName`.

File: src/backend/services/mcp/local-claude-orchestrator.ts [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+      await this.serializeMcpServers(manager, options.serverFilter);
+
+    if (skippedNonWhitelistedServers.length > 0) {
+      console.warn(
+        `[LocalClaudeOrchestrator] No whitelisted tools for server(s): ${skippedNonWhitelistedServers.join(", ")} under "${approvalMode}" approval mode — proceeding with whatever is whitelisted; non-whitelisted tools were skipped.`,
+      );
+    }
+
+    const tmpConfigPath = join(tmpdir(), `yakshaver-mcp-${randomUUID()}.json`);
+    await fs.writeFile(tmpConfigPath, JSON.stringify(mcpConfig, null, 2), "utf8");
```
Comment (calumjs) [inline src/backend/services/mcp/local-claude-orchestrator.ts]: MEDIUM: This writes the serialized MCP config — including the live `Authorization: Bearer <oauth access token>` injected at line 268 — to the OS-shared temp dir with default permissions and no `mode`. On a multi-user machine the token is readable by other local users, and since cleanup is only in a `finally` (`fs.unlink`), a SIGKILL leaves it on disk. This bypasses the app convention of encrypting all secrets at rest via Electron `safeStorage` (`base-secure-storage.ts`). Prefer passing the config inline via `--mcp-config <json>` so the token never touches disk, or write with `{ mode: 0o600 }` plus cleanup on exit/signal.

File: src/backend/services/mcp/local-claude-orchestrator.ts (hunk 1 above)
Comment (calumjs) [reply src/backend/services/mcp/local-claude-orchestrator.ts]: Fixed in 4983574. The tool_result branch no longer records the opaque `tool_use_id` as the tool name. `handleStreamEvent` now builds an id->name map from the assistant `tool_use` blocks (which carry both `id` and `name`) and resolves `tool_use_id` back to the human-readable name on the result, falling back to the id only if the tool_use block was never seen. This matches the OpenAI twin (`mcp-orchestrator.ts` records `toolCall.toolName`), so the toolActivity fed into the #833 judge prompt and the diagnostics log now carry e.g. `mcp__GitHub__create_issue` instead of `toolu_...`. The fixtures now use realistic `toolu_...` ids on both the tool_use `id` and the tool_result `tool_use_id`, and the test asserts the judge prompt contains the real name and NOT the opaque id (the prior fixture's name-shaped id masked the bug).

File: src/backend/services/mcp/local-claude-orchestrator.ts (hunk 2 above)
Comment (calumjs) [reply src/backend/services/mcp/local-claude-orchestrator.ts]: Fixed in 4983574. The temp `--mcp-config` (which carries the live `Authorization: Bearer <oauth token>` for each HTTP server) is now written owner-only with `{ mode: 0o600 }`, so it is no longer readable by other local users in the shared system temp dir. I kept the file-on-disk approach (rather than inline `--mcp-config '<json>'`) deliberately: passing the token in argv would expose it in the process list (`ps` / Task Manager) to other local users, which is a worse exposure on a shared machine than a 0o600 file. On Windows the POSIX mode is a no-op but the per-user temp dir is already ACL-restricted there. Cleanup remains in `finally`.

**Review by calumjs — COMMENTED** [review]
## Review summary

This PR adds a new `yakshaver` CLI (`src/cli/`), a localhost HTTP control-plane bridge (`src/backend/services/cli-bridge/`), and a `zod` runtime dependency. The functionality is coherent, but a few issues are worth addressing before merge — two around the project's documented security/docs contract, and two around input validation / config normalization in the bridge.

### High

**1. AGENTS.md not updated despite Rule 6 being squarely triggered**
`src/cli/` (+ the `yakshaver` bin), `src/backend/services/cli-bridge/`, the new localhost HTTP bridge pattern, and the new `zod` root dependency all fall under AGENTS.md Rule 6 ("Keep AGENTS.md Up to Date", AGENTS.md:174-176), which requires updating the doc in the same PR when a change touches dependencies, patterns, directory structure, or workflows. The diff modifies zero `.md` files, so the Directory Tree (AGENTS.md:77-141, no `src/cli`/`cli-bridge`), Key Dependencies (AGENTS.md:50-58, no `zod` as a root runtime dep), and Architecture Patterns (which document IPC as the only frontend↔backend channel, no bridge) still describe the pre-PR state.
Remedy: in this PR, add `src/cli/` and `src/backend/services/cli-bridge/` to the Directory Tree, add `zod` to Key Dependencies, and document the localhost CLI-bridge as a new Architecture Pattern plus the `yakshaver` CLI as a feature.

**2. Plaintext bearer token written into the `yakshaver-tokens/` directory the docs reserve for encrypted files**
`writeTokenFile` (cli-bridge-server.ts:1010-1023) writes the bridge's sole auth credential as plaintext JSON to `userData/yakshaver-tokens/cli-bridge.json` (mode 0o600, no safeStorage). This contradicts the documented secure-storage contract: AGENTS.md:369 ("Never store API keys, tokens, or credentials in plain text"), AGENTS.md:636-637 (safeStorage / BaseSecureStorage subclasses for all credentials), and AGENTS.md:582 + README.md:37-41 which document `yakshaver-tokens/` as holding only safeStorage-encrypted `*.enc` files. Every other token (LlmStorage, GitHubTokenStorage, McpOAuthTokenStorage) goes through BaseSecureStorage.
A plaintext handshake token may be a defensible choice (the CLI is a plain Node process that can't use Electron safeStorage), but then the contract must be reconciled: either encrypt via a BaseSecureStorage subclass so the directory stays all-encrypted, or move the plaintext file to a distinct, clearly-named location and update AGENTS.md:582 + README.md:37-41 to document the exception (0o600, localhost-only) and why it is acceptable.

### Medium

**3. CLI hardcodes `YakShaver`/`YakShaverDev` userData folder while docs say `SSW.YakShaver`**
`getUserDataDir` (user-data-path.ts:57-70) hardcodes the Electron app name `YakShaver` / `YakShaverDev`, which is correct (electron-builder.config.js:3, package.json productName; dev override at src/backend/index.ts:47-49). But README.md:39-41 and AGENTS.md:578-582 say the folder is `SSW.YakShaver`. This PR codifies the real name in code yet leaves the docs contradicting it, so a reviewer following the docs would conclude the CLI points at the wrong directory.
Remedy: correct README.md:39-41 and AGENTS.md:578-582 to `YakShaver` (noting the `YakShaverDev` dev override) as part of this PR.

**4. `POST /llm/config` skips the safeParse validation every other write path uses**
`routeLlm` POST (bridge-router.ts:667-678) does only `if (config.version !== 2)` then casts `req.body as LLMConfigV2` and hands it straight to `storeLLMConfig` (llm-storage.ts:72), which encrypts and stores without validation. Every other mutating endpoint in the same file (POST/PUT `/mcp/servers`, `/enabled`, `/llm/config/orchestrator`, PATCH `/settings`) validates the body via a shared zod schema with `safeParse`. A `version:2` body with garbage/partial/wrong-typed fields (e.g. a bad `languageModel`) is persisted as-is and breaks downstream (language-model-provider.ts indexes `LLM_PROVIDER_CONFIGS[provider]` and passes `apiKey` blindly). Not reachable via the shipped CLI (commands.ts blocks `config set llm`), but the bridge endpoint is open to any localhost caller holding the token, and the malformed-v2 branch is untested.
Remedy: add an `LLMConfigV2` zod schema and `safeParse` `req.body` (400 on failure) like the other endpoints; or drop the raw `POST /llm/config` write and keep only the validated orchestrator endpoint.

**5. `mcp update --transport` persists mixed HTTP/stdio configs with stale fields**
Changing transport via `mcp update` leaves stale fields from the previous transport. The CLI maps `--transport` into the patch (commands.ts:173-180); the bridge merges via a blind spread `{ ...existing, ...parsed.data, id }` (bridge-router.ts:152-157) with `McpServerPatchSchema` being a flat object (not a discriminated union, protocol.ts) where url/headers and command/args/env are all optional; `MCPServerManager.updateServerAsync` repeats the spread (mcp-server-manager.ts:298-302); and `validateServerConfig` (mcp-server-manager.ts:391-411) only checks the required field for the selected transport without clearing the other's. Result: `yakshaver mcp update <httpId> --transport stdio --command node` keeps the old `url`/`headers`, and `printServerLine` (cli/index.ts:124,129 — prefers `s.url ?? s.command`) then prints the stale URL instead of the new command. No transport-change test exists in bridge-router.test.ts.
Remedy: when transport changes, normalize to the selected union member (clear the other transport's fields), require the selected transport's field in the same update (or reject), and add tests for HTTP→stdio and stdio→HTTP.

File: src/backend/services/cli-bridge/bridge-router.ts [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+      // and overlay them on the existing server. The server must already exist.
+      const parsed = McpServerPatchSchema.safeParse(req.body);
+      if (!parsed.success) {
+        return fail(`Invalid server config: ${formatZodError(parsed.error)}`);
+      }
+      const existing = await services.mcp.getServerByIdAsync(serverId);
+      if (!existing) {
+        return fail(`MCP server '${serverId}' not found`, 404);
+      }
+      const merged = {
```
Comment (calumjs) [inline src/backend/services/cli-bridge/bridge-router.ts]: Medium: this blind merge `{ ...existing, ...parsed.data, id }` lets `mcp update --transport` persist mixed HTTP/stdio configs — switching HTTP->stdio keeps the old `url`/`headers` (McpServerPatchSchema is a flat object, not a discriminated union). `updateServerAsync` (mcp-server-manager.ts:298-302) repeats the spread and `validateServerConfig` (391-411) only checks the selected transport's required field. Normalize to the selected union member (clear the other transport's fields) and add HTTP<->stdio tests.

File: src/backend/services/cli-bridge/cli-bridge-server.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+    const payload: CliBridgeTokenFile = {
+      port: this.port ?? CLI_BRIDGE_DEFAULT_PORT,
+      token: this.token ?? "",
+      startedAt: new Date().toISOString(),
+      version: safeAppVersion(),
+    };
+
+    // Owner-only permissions where the platform honours them (no-op on Windows).
+    await fs.writeFile(filePath, JSON.stringify(payload, null, 2), { mode: 0o600 });
```
Comment (calumjs) [inline src/backend/services/cli-bridge/cli-bridge-server.ts:130]: High: this writes the bridge's sole auth credential as plaintext JSON to `userData/yakshaver-tokens/cli-bridge.json`. AGENTS.md:582 + README.md:37-41 document that directory as holding only safeStorage-encrypted `*.enc` files, and AGENTS.md:369,636-637 require BaseSecureStorage/safeStorage for all credentials. Either encrypt via a BaseSecureStorage subclass so the directory stays all-encrypted, or move this plaintext file to a distinct location and update the docs to note the localhost-only, 0o600 exception.

File: src/cli/user-data-path.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  if (process.platform === "darwin") {
+    return join(homedir(), "Library", "Application Support");
+  }
+  return process.env.XDG_CONFIG_HOME ?? join(homedir(), ".config");
+}
+
+/** Resolve the YakShaver userData directory. Pass `dev: true` for a dev build. */
+export function getUserDataDir(dev = isDevEnv()): string {
+  const appName = dev ? "YakShaverDev" : "YakShaver";
```
Comment (calumjs) [inline src/cli/user-data-path.ts:31]: Medium: this hardcodes the real app name (`YakShaver`/`YakShaverDev`) correctly, but README.md:39-41 and AGENTS.md:578-582 still say the userData folder is `SSW.YakShaver`. Update those docs to `YakShaver` (noting the `YakShaverDev` dev override at src/backend/index.ts:47-49) so docs and code agree.

File: src/backend/services/cli-bridge/bridge-router.ts [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+        r.orchestrationBackend = DEFAULT_ORCHESTRATION_BACKEND;
+      }
+    }
+    return ok(redacted);
+  }
+  if (req.method === "POST") {
+    if (!req.body || typeof req.body !== "object") {
+      return fail("Invalid LLM config payload");
+    }
+    const config = req.body as LLMConfigV2;
```
Comment (calumjs) [inline src/backend/services/cli-bridge/bridge-router.ts]: Medium: `POST /llm/config` casts `req.body as LLMConfigV2` after only a `version !== 2` check, then hands it to `storeLLMConfig` (no validation). Every other write path in this file uses a shared zod `safeParse`. A malformed/wrong-typed version:2 body is persisted as-is and breaks downstream consumers (language-model-provider indexes `LLM_PROVIDER_CONFIGS[provider]` and passes `apiKey` blindly). Add an `LLMConfigV2` zod schema + safeParse (400 on failure), or drop the raw write and keep only the validated orchestrator endpoint.

File: src/backend/services/cli-bridge/bridge-router.ts (hunk 3 above)
Comment (calumjs) [reply src/backend/services/cli-bridge/bridge-router.ts]: Fixed in 0fa1d88. `PUT /mcp/servers/:id` now normalizes to the target transport's union member via a new `mergeMcpPatch` helper: when the transport changes it strips the other transport's fields (HTTP -> stdio drops `url`/`headers`/`version`/`timeoutMs`; stdio -> HTTP drops `command`/`args`/`env`/`cwd`/`stderr`) and requires the new transport's mandatory field (`url` / `command`) in the same patch or already on the existing config, otherwise the update is rejected with a 400. So no stale fields survive a switch and the CLI prints the correct target. Added bridge-router tests for both directions plus the missing-required-field rejections.

File: src/backend/services/cli-bridge/cli-bridge-server.ts (hunk 4 above)
Comment (calumjs) [reply src/backend/services/cli-bridge/cli-bridge-server.ts:130]: Addressed in 0fa1d88 via the documented-exception remedy. The plaintext token is intrinsic here: the standalone `yakshaver` CLI is a plain Node process and cannot use Electron `safeStorage` to decrypt a handshake token, so encrypting it would break the CLI. I documented the exception instead: AGENTS.md Storage Locations now lists `yakshaver-tokens/cli-bridge.json` as a plaintext JSON handshake file (`0o600`, localhost-only, regenerated each start), and the Security section's safeStorage rule now spells out this one exception and why it's acceptable. The directory's `*.enc` files remain the encrypted-only contract; `cli-bridge.json` is the explicitly-noted exception.

File: src/cli/user-data-path.ts (hunk 5 above)
Comment (calumjs) [reply src/cli/user-data-path.ts:31]: Fixed in 0fa1d88. README.md and AGENTS.md Storage Locations now use the real folder name `YakShaver` (verified against `%APPDATA%` on disk: `YakShaver`/`YakShaverDev`, no `SSW.YakShaver`), with a note that dev builds use `YakShaverDev` (referencing the override at src/backend/index.ts:47-49). Docs and `getUserDataDir` now agree.

File: src/backend/services/cli-bridge/bridge-router.ts (hunk 6 above)
Comment (calumjs) [reply src/backend/services/cli-bridge/bridge-router.ts]: Fixed in 0fa1d88. Added a shared `LLMConfigV2Schema` (strict zod, mirroring `LLMConfigV2` incl. the per-provider `ModelConfig` discriminated union) to `src/shared/cli-bridge/protocol.ts`, and `POST /llm/config` now `safeParse`s the body and returns 400 on failure instead of the bare `as` cast — matching every other mutating endpoint. A malformed/wrong-typed or unknown-field v2 body is now rejected before it reaches `storeLLMConfig`. Added tests for a wrong-typed `languageModel` and unknown top-level fields.

**Review by calumjs — COMMENTED** [review]
## Review summary

This change is functionally close but has a few correctness/UX defects on edge paths plus a PR-hygiene problem. Findings grouped by severity below. Where the original line references pointed at `gh pr diff` output offsets, I've noted the true file location.

---

### High

**1. Disabling/mutating a built-in MCP server via the CLI silently writes shadow junk and reports false success**
`src/backend/services/cli-bridge/bridge-router.ts` (enabled route) + `src/backend/services/mcp/mcp-server-manager.ts:220,288-316,372-417`

`yakshaver mcp enable <builtin-id> --off` (and the `--name <builtin>` variant) resolves a built-in server (e.g. `yak_video_tools`, transport `inMemory`) because `getServerByIdAsync` -> `getAllServerConfigsAsync` merges internal servers, so the 404 guard passes. The route then calls `updateServerAsync(builtinId, {...existing, enabled:false})`. Since the builtin id is not in the stored list, `updateServerAsync` PUSHES a stored copy; `validateServerConfig` does not reject an `inMemory` config, so the junk is persisted. But `mergeWithInternalServers` force-sets `enabled = true` for internal servers on every read and skips stored entries with `builtin:true` (`if (s.builtin) continue`). Net: the disable is ineffective, a dead shadow copy lands in `mcp-servers.enc` that is never surfaced or cleaned up, and the CLI prints "Disabled MCP server" — a false success. No test covers a builtin/inMemory id in the enabled toggle.

Remedy: in the enabled/update/remove routes, detect configs carrying `builtin: true` and reject the mutation with a clear error ("'<name>' is a built-in server and cannot be modified/removed via the CLI"), or have `updateServerAsync` refuse to persist a config whose id matches a builtin. Not remotely exploitable (bridge binds 127.0.0.1 + bearer token), but a real correctness/UX defect plus persisted junk.

---

### Medium

**2. Transport-switch field-strip in `mergeMcpPatch` is silently undone by `updateServerAsync`**
`src/backend/services/cli-bridge/bridge-router.ts` (`mergeMcpPatch`) vs `src/backend/services/mcp/mcp-server-manager.ts:288-316,372-417`

On an HTTP->stdio (or reverse) switch, `mergeMcpPatch` does `delete merged[field]` to strip the old transport's fields, then the PUT handler passes that stripped config to `updateServerAsync`. The real `updateServerAsync` re-fetches `existing` from storage and does `const merged = { ...existing, ...config, id }`. Because the stripped `config` has no `url`/`headers` key to override them, the stale fields from `existing` are spread back in, and `validateServerConfig` never rejects foreign keys. Result: a `stdio` server is persisted carrying leftover `url`/`headers` (or vice-versa) — violating the discriminated-union contract and the PR's own design comment ("no stale url/headers survive a switch"). The stale `url` is dead data at runtime (client construction dispatches on `transport`), but stale `headers` can retain Authorization bearer tokens for a server that no longer uses HTTP — a data-hygiene regression.

The "strips stale" tests mock `updateServerAsync` and assert on the value passed in; they never run the manager's second merge. `cli-bridge-server.test.ts` stubs `MCPServerManager` to `{}`. So no test exercises the strip->real-persist path.

Remedy: treat a transport change as a full replace (have `updateServerAsync` replace rather than overlay, or have the bridge pass the complete normalized target config), and add an integration test driving `mergeMcpPatch` through the REAL `MCPServerManager.updateServerAsync` (with `McpStorage` mocked) asserting no `url`/`headers` remain after an HTTP->stdio switch.

**3. `config get orchestrator` prints `null` instead of `openai` on a fresh install**
`src/backend/services/cli-bridge/bridge-router.ts` (routeLlm GET default-injection guard)

The handler injects the `openai` default only when `redacted && typeof redacted === 'object'`. On a fresh install `getLLMConfig()` returns `null` (`llm-storage.ts:82-84`), `redactLlmConfig(null)` returns `null` (`redact.ts:45`), the guard is false, and `ok(null)` is returned — so `config get orchestrator` prints `null`, contradicting the handler's own comment ("always shows a concrete value even when the field has never been set"). The CLI applies no default of its own. The "defaults orchestrationBackend to openai when unset" test only covers a non-null config missing the field; the null path is untested.

Remedy: apply `DEFAULT_ORCHESTRATION_BACKEND` before the object typeguard (or return a minimal `{ orchestrationBackend: DEFAULT_ORCHESTRATION_BACKEND }` on null), and add a test with `getLLMConfig` mocked to `null`.

**4. CLI `config set settings --open-at-login` does not take effect for the current session**
`src/backend/services/cli-bridge/bridge-router.ts` (routeSettings PATCH) vs `src/backend/ipc/user-settings-handlers.ts:32,98,137,145`

The bridge PATCH route calls `UserSettingsStorage.updateSettingsAsync(patch)` directly, which only persists + updates the in-memory cache. It bypasses the IPC handler's `handleOpenAtLoginUpdate()` — the only place outside startup that calls `app.setLoginItemSettings()`. So a CLI `--open-at-login true` persists the flag and reports "Updated settings", but the OS launch-item is not registered until the next app restart (startup `syncLoginItemSettings` re-applies it) — a behaviour divergence from the UI toggle a user would expect to be equivalent. No test covers an `openAtLogin` PATCH or its side effect.

Remedy: route bridge settings mutations through a shared `applySettingsUpdate()` used by both the IPC handler and the bridge (so side effects like `setLoginItemSettings` fire), or at minimum document that `openAtLogin` via CLI takes effect on next restart.

**5. Brittle space-split of stdio `--args` shreds any argument containing a space**
`src/cli/commands.ts:67,190`

`--args` is parsed with `value.split(" ").filter(Boolean)` with no quoting/escape mechanism, so a single argument that legitimately contains a space (a path like `/My Documents/server.js`, a JSON literal, a quoted flag value) is silently split into multiple argv entries. Such servers cannot be configured via the CLI at all, and the failure is silent — the CLI returns success while the spawned stdio server gets the wrong argv. The existing test cements the lossy behaviour (`--args "server.js --port 3000"` -> `["server.js","--port","3000"]`) with no coverage of an intentional internal space.

Remedy: accept `--arg` repeatably (one value per flag, preserving spaces), or a JSON array, or a quote-aware tokenizer. At minimum reject ambiguous input rather than silently splitting.

**6. PR title and body are malformed**
PR #914 metadata

The title carries a stray corrupt prefix `L⚠️ ◾` before the valid subject, and the body is the literal string `@-` (the classic `gh api -f body=@-` stdin gotcha — the flag wrote the placeholder instead of reading a file). A reviewer/merger gets no what/why, no acceptance criteria, no test plan, and since a squash merge takes the PR title as the commit subject, the corrupt prefix would land in permanent history alongside the repo's Conventional-Commits convention.

Remedy: re-set the title to a clean subject (e.g. `feat(cli): orchestration backend setting, mcp update, --name lookups (#913)`) and replace the body with a real description (problem, approach, security model, test plan). When scripting via `gh`, pipe the body with `--body-file -` or `-F body=@file`, not `-f body=@-`.
